src/pages/Browse/index.spec.tsx

```diff
@@ -542,4 +548,157 @@ describe(__filename, () => {
+      }
+      if (beginFetchNextFile) {
+        store.dispatch(
+          versionsActions.beginFetchVersionFile({
+            path: file2,
+            versionId: version.id,
+          }),
+        );
+      }
+      if (loadNextFile) {
+        store.dispatch(
+          versionsActions.loadVersionFile({ path: file2, version }),
+        );
+      }
+      const dispatchSpy = spyOn(store, 'dispatch');
+
+      render({
+        _fetchVersionFile,
+        addonId: String(addonId),
+        store,
+        versionId: String(version.id),
+      });
+
+      return {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        store,
+        version,
+      };
+    };
+
+    it('dispatches fetchVersionFile for the next file', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        fakeThunk,
+        version,
+        dispatchSpy,
+      } = setUpFilesAndRender();
+
+      expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).toHaveBeenCalledWith({
+        addonId,
+        path: file2,
+        versionId: version.id,
+      });
+    });
+
+    it('does not dispatch fetchVersionFile for the next file if the current version file is not loaded', () => {
+      const { _fetchVersionFile, addonId, version } = setUpFilesAndRender({
+        loadVersionFile: false,
+      });
+
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+        addonId,
+        path: file2,
+        versionId: version.id,
+      });
+    });
+
+    it('does not dispatch fetchVersionFile for the next file when fileTree is not built', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        version,
+      } = setUpFilesAndRender({
+        buildTree: false,
+      });
+
+      expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+        addonId,
+        path: file2,
+        versionId: version.id,
+      });
+    });
+
+    it('does not dispatch fetchVersionFile for the next file when the next file is loading', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        version,
+      } = setUpFilesAndRender({ beginFetchNextFile: true });
+
+      expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
+        addonId,
+        path: file2,
+        versionId: version.id,
+      });
+    });
+
+    it('does not dispatch fetchVersionFile when the next file is already loaded', () => {
+      const {
+        _fetchVersionFile,
+        addonId,
+        dispatchSpy,
+        fakeThunk,
+        version,
+      } = setUpFilesAndRender({
+        loadNextFile: true,
+      });
+
+      expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
+      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
```

Comment:
It looks like this (and maybe others?) can be simplified to just checking for any dispatch at all:

```diff
diff --git a/src/pages/Browse/index.spec.tsx b/src/pages/Browse/index.spec.tsx
index 414e1ac..5aecc16 100644
--- a/src/pages/Browse/index.spec.tsx
+++ b/src/pages/Browse/index.spec.tsx
@@ -694,11 +694,7 @@ describe(__filename, () => {
       });
 
       expect(dispatchSpy).not.toHaveBeenCalledWith(fakeThunk.thunk);
-      expect(_fetchVersionFile).not.toHaveBeenCalledWith({
-        addonId,
-        path: file2,
-        versionId: version.id,
-      });
+      expect(_fetchVersionFile).not.toHaveBeenCalled();
     });
   });
 });
```